pkg/config/config.go

```diff
@@ -30,9 +32,25 @@ type Info interface {
 type ComponentSettings struct {
 
 	// The builder image to use
-	ComponentType *string `yaml:"ComponentType,omitempty"`
+	Type *string `yaml:"Type,omitempty"`
+
+	// Path is path to binary in current/context dir
+	Path *string `yaml:"Path,omitempty"`
+
+	// Ref is component source git ref but can be levaraged for more in future
+	Ref *string `yaml:"Ref,omitempty"`
+
+	// Type is type of component source: git/local/binary
+	SourceType *occlient.CreateType `yaml:"SourceType,omitempty"`
+
+	// Ports is a slice of ports to be exposed when a component is created
+	Ports *[]string `yaml:"Ports,omitempty"`
```

Comment:
shouldn't be port `int32`?

Reply:
Ports I think are specified as for ex: 8080/TCP

Reply:
then it needs to be explained in the comment. It is not obvious.

Reply:
so should I add that in the comment?

Reply:
yes, if you are touching some code, and you see something wrong fix it right away. Otherwise, it might be never fixed.

Reply:
yep

Reply:
added comment